### PR TITLE
Gracefully handle create requests for existing offline tables.

### PR DESCRIPTION
**Pull request by antumbde**
Return 409 Conflict for requests to create existing offline tables.
POST for existing realtime tables is allowed and handled gracefully because
our tooling relies on that right now for migration to LLC.

### CODE REVIEW

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java [hunk 1]
```diff
@@ -71,6 +71,15 @@ public void testCreateTable() throws Exception {
         "MMAP", "v1");
 
     sendPostRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forTableCreate(), request.toString());
+    boolean tableExistsError = false;
+    try {
+      sendPostRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forTableCreate(), request.toString());
+    } catch (IOException e) {
+      System.out.println(e.getMessage());
```
Comment (mcvsubbu) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java]: pls remove system.out.println

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java [hunk 2]
```diff
@@ -71,6 +71,15 @@ public void testCreateTable() throws Exception {
...
+      Assert.assertTrue(e.getMessage().startsWith("Server returned HTTP response code: 409"));
```
Comment (mcvsubbu) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java]: you can print the actual message here if assert failed.

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java (hunk 1 above)
Comment (antumbde) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java]: fixed

File: pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java (hunk 2 above)
Comment (antumbde) [inline pinot-controller/src/test/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResourceTest.java]: fixed

**Pull request merged**